packages/daemon/src/worker.js

```diff
@@ -55,12 +50,11 @@ export const makeWorkerFacet = ({
     },
 
     /**
-     * @param {string} path
+     * @param {string} specifier
      * @param {unknown} powersP
      */
-    importUnsafeAndEndow: async (path, powersP) => {
-      const url = pathToFileURL(path);
-      const namespace = await import(url);
+    importUnsafeAndEndow: async (specifier, powersP) => {
+      const namespace = await import(specifier);
```

Comment:
Here I suppose you're putting `specifier` into `import`, so you don't need to resolve it first, so this isn't all that relevant, but...

Based on a clue from @michaelfig , I have migrated from using `resolve` on `path` to `createRequire(...)`.
For example:

https://github.com/Agoric/dapp-offer-up/blob/58d5ee78e302bfbcb2696375be57d9a19a0fca41/contract/test/test-bundle-source.js#L9-L15

Reply:
Interesting. By way of comparison:

- `path.resolve` will take a path and create a fully qualified path, which Node.js will accept as a module specifier for purposes of `require` or `import`.
- `createRequire(import.meta.url)` will give you a `require.resolve` which accepts CommonJS module specifiers and so distinguishes `./local-path.js` from 'in-your-node-modules` and uses the CommonJS module loader semantics.
- `import.meta.resolve` is intended to replace `require.resolve` and surface the Node.js 16+ ESM module resolution, respecting `"exports"` and `"imports"` in `package.json`. It’s shimmed with the `import-meta-resolve` package in npm, but that’s async. When `import.meta.resolve` is broadly supported, it will be synchronous because it benefits from the `package.json` cache Node.js builds while loading the surrounding module.

For this particular case, the Endo Pet Dæmon needs a module specifier as specific as a fully qualified path `file:///` URL since the Dæmon and the plugin are installed in separate locations and should have completely disjoint `package.json` graphs. So, there’s no expectation that the pet dæmon will be able to figure out where `@dckc/plugin` was installed.